Modify Copy and Remove button behaviors

howdyjessie opened the pull request:
This PR brings two updates:
- #433: The copy button duplicates only the most recently added row/column in the data entry/recordedit app. But before it attempts the duplication, the app first checks that the most recently added row/column doesn't have any invalid values entered by the user.
- The delete button that's used to delete an individual row/column from the data entry form is now displayed via a new Chaise-wide `deleteLink` directive. This directive displays a UI control that performs a deletion routine that you pass in. If the `confirmDelete` property in `chaise-config.js` is `true`, the directive will display a confirmation modal before running your deletion routine. More details about usage can be found at the top of [`delete-link.js`](https://github.com/informatics-isi-edu/chaise/blob/recedit-copy-row/common/delete-link.js#L1-L22).

**To Test**
These changes can be seen in my [user dir](https://dev.isrd.isi.edu/~jessie/chaise/recordedit/#1/legacy:dataset). The code can be pulled down from this branch.

Tagging @shinyichen and @jrchudy so they know about this new directive.

hongsudt commented:
@howdyjessie I need some training to get me started with how to review the pull request..

howdyjessie commented:
@hongsudt No problem, it's not difficult to get started! I can swing by your office when you've got some spare time.

howdyjessie commented:
@hongsudt I fixed the issue that caused the test URL to break. (It was because I forgot my user dir's files reflected the files needed for a separate PR but not this one.) I've updated the [test URL](https://dev.isrd.isi.edu/~jessie/chaise/recordedit/#1/legacy:dataset) at the top of this thread to reflect data entry's name change to recordedit. Can you try again?
